Modify the ESLint config to allow .js ext for React component file

cometkim opened the pull request:
#### Summary
Modify the ESLint setups to allow .js ext for React component file.

@saturninoabril noticed me change about JavaScript filename extensions in https://github.com/mattermost/mattermost-webapp/pull/1666#pullrequestreview-163366241, but the ESLint settings does not allow `.js` for react yet.

hmhealey commented:
@cometkim You're going to need to redo your changes. I actually just moved most of the ESLint config into a shared repository, so most of that file has been removed, except not the line you're changing

cometkim commented:
@hmhealey I got it. rebased onto latest config, changed to allow only '.js' ext and to set level 'warn'

I also can change it to 'error' level and change every filenames in this PR.

cometkim commented:
rebased.

hmhealey commented:
Thanks!

hmhealey commented:
I've got a task to rename all of the files to js at some point and disable jsx files, but I'm going to wait until after hacktoberfest and our 5.5 quality release is mostly done
